great_expectations/core/util.py

```diff
@@ -408,3 +408,112 @@ def url(self):
 def sniff_s3_compression(s3_url: S3Url) -> str:
     """Attempts to get read_csv compression from s3_url"""
     return _SUFFIX_TO_PD_KWARG.get(s3_url.suffix, "infer")
+
+
+# noinspection PyPep8Naming
+def get_or_create_spark_application(
+    spark_config: Optional[Dict[str, str]] = None,
+):
+    # Due to the uniqueness of SparkContext per JVM, it is impossible to change SparkSession configuration dynamically.
+    # Attempts to circumvent this constraint cause "ValueError: Cannot run multiple SparkContexts at once" to be thrown.
+    # Hence, SparkSession with SparkConf acceptable for all tests must be established at "pytest" collection time.
+    # This is preferred to calling "return SparkSession.builder.getOrCreate()", which will result in the setting
+    # ("spark.app.name", "pyspark-shell") remaining in SparkConf statically for the entire duration of the "pytest" run.
+    try:
+        from pyspark import SparkContext
+        from pyspark.sql import SparkSession
+    except ImportError:
+        SparkContext = None
+        SparkSession = None
+        # TODO: review logging more detail here
+        logger.debug(
+            "Unable to load pyspark; install optional spark dependency for support."
+        )
+
+    if spark_config is None:
+        spark_config = {
+            "spark.sql.catalogImplementation": "hive",
+            "spark.executor.memory": "450m",
```

Comment:
Very minor -- let's have our default be empty

Reply:
@jcampbell : Done -- please re-review.  Thank you.